rd_server.py

```diff
@@ -102,46 +121,64 @@ def run(self):
         run.runid = run_id
         run.rundir = config['runs'] + '/' + run_id
         run.log = log_file
-        run.set = info['task']
-        if 'arch' in info:
-            run.arch = info['arch']
+        if type(info['ctcSets']) is list:
+            run_set_list = info['ctcSets']
         else:
-            run.arch = 'x86_64'
-        run.bindir = run.rundir + '/' + run.arch + '/'
-        run.prefix = run.rundir + '/' + run.set
-        try:
-            os.mkdir(run.prefix)
-        except FileExistsError:
-            pass
-        if 'qualities' in info:
-          if info['qualities'] != '':
-              run.quality = info['qualities'].split()
-        if 'extra_options' in info:
-          run.extra_options = info['extra_options']
-        if 'save_encode' in info:
-            if info['save_encode']:
-                run.save_encode = True
-        run.status = 'running'
-        run.write_status()
-        run_list.append(run)
-        video_filenames = video_sets[run.set]['sources']
-        run.set_type = video_sets[run.set].get('type', 'undef')
-        run.work_items = create_rdwork(run, video_filenames)
-        work_list.extend(run.work_items)
-        if False:
-            if 'ab_compare' in info:
-                if info['ab_compare']:
-                    abrun = ABRun(info['codec'])
-                    abrun.runid = run_id
-                    abrun.rundir = config['runs'] + '/' + run_id
-                    abrun.log = log_file
-                    abrun.set = info['task']
-                    abrun.bindir = config['codecs'] + '/' + info['codec']
-                    abrun.prefix = run.rundir + '/' + run.set
-                    run_list.append(abrun)
-                    abrun.work_items.extend(create_abwork(abrun, video_filenames))
-                    work_list.extend(abrun.work_items)
-                    pass
+            if info['ctcSets'] == 'aomctc-all':
```

Comment:
would be nice to possibly split aomctc-all and aomctc-mandatory into the list of subsets in the frontend before it reaches rd_server, to result in less duplication of these lists

Reply:
If you did this, you'd probably have to change the json too...